Sources_v3/StreamChatUI/ChatChannel/ChatFileAttachmentListView.swift

```diff
@@ -0,0 +1,52 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+open class ChatFileAttachmentListView<ExtraData: ExtraDataTypes>: View, UIConfigProvider {
+    public var itemHeight: CGFloat = 56
+
+    public var data: AttachmentListViewData<ExtraData>? {
+        didSet { updateContentIfNeeded() }
+    }
+
+    // MARK: - Subviews
+
+    public private(set) lazy var stackView: UIStackView = {
+        let stack = UIStackView()
+        stack.isLayoutMarginsRelativeArrangement = true
+        stack.preservesSuperviewLayoutMargins = true
+        stack.distribution = .fillEqually
+        stack.axis = .vertical
+        return stack.withoutAutoresizingMaskConstraints
+    }()
+
+    // MARK: - Overrides
+
+    override public func defaultAppearance() {
+        stackView.spacing = 4
+        directionalLayoutMargins = .init(top: 4, leading: 4, bottom: 4, trailing: 4)
+    }
+
+    override open func setUpLayout() {
+        embed(stackView)
+    }
+
+    override open func updateContent() {
+        stackView.arrangedSubviews.forEach {
+            stackView.removeArrangedSubview($0)
+            $0.removeFromSuperview()
```

Comment:
nit: `removeFromSuperview` should be enough to remove view from stackview.

> When a subview is removed from the stack view, the stack view also removes it from the arrangedSubviews array.

Reply:
fixed in 69d1d39f9e5f8ced3bfd9d96af2dc4f530ee8192